packages/lib-classifier/src/store/UserProjectPreferencesStore.spec.js

```diff
@@ -0,0 +1,249 @@
+import sinon from 'sinon'
+import asyncStates from '@zooniverse/async-states'
+import { getEnv, types } from 'mobx-state-tree'
+
+import ProjectStore from './ProjectStore'
+import RootStore from './RootStore'
+import UserProjectPreferencesStore from './UserProjectPreferencesStore'
+import {
+  ProjectFactory,
+  UPPFactory,
+  UserFactory
+} from '../../test/factories'
+import stubPanoptesJs from '../../test/stubPanoptesJs'
+
+let rootStore
+const project = ProjectFactory.build()
+const upp = UPPFactory.build()
+const user = UserFactory.build()
+const token = '1234'
+
+
+const clientStub = stubPanoptesJs({
+  projects: project,
+  project_preferences: upp
+})
+
+const clientStubWithoutUPP = stubPanoptesJs({
+  projects: project,
+  project_preferences: null
+})
+
+const authClientStubWithoutUser = {
+  checkCurrent: sinon.stub().callsFake(() => Promise.resolve(null)),
+  checkBearerToken: sinon.stub().callsFake(() => Promise.resolve(null))
+}
+
+const authClientStubWithUser = {
+  checkCurrent: sinon.stub().callsFake(() => Promise.resolve(user)),
+  checkBearerToken: sinon.stub().callsFake(() => Promise.resolve(token))
+}
+
+describe.only('Model > UserProjectPreferencesStore', function () {
+  it('should exist', function () {
+    expect(UserProjectPreferencesStore).to.be.an('object')
+  })
+
+  it('should remain in an initialized state if there is no project', function () {
+    rootStore = RootStore.create({
+      projects: ProjectStore.create(),
+      userProjectPreferences: UserProjectPreferencesStore.create()
+    }, { authClient: authClientStubWithoutUser, client: clientStub })
+
+    expect(rootStore.userProjectPreferences.loadingState).to.equal(asyncStates.initialized)
+  })
+
+  it('should set project preferences if there is a user and a project', function (done) {
+    rootStore = RootStore.create({
+      projects: ProjectStore.create(),
+      userProjectPreferences: UserProjectPreferencesStore.create()
+    }, { authClient: authClientStubWithUser, client: clientStub })
+    rootStore.projects.setActive(project.id)
+      .then(() => {
+        const uppInStore = rootStore.userProjectPreferences.active
+        expect(uppInStore.toJSON()).to.deep.equal(upp)
+      }).then(done())
```

Comment:
This passes the return value of `done()` to `then` as a callback function. It should be `.then(done, done)` to catch failing tests.

Reply:
Thanks for catching that. I usually return the promise, but using the done callback just made more sense here.